### PR TITLE
Update to builder 0.4.0, otelcol to 0.18.0

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ ifeq (, $(shell which opentelemetry-collector-builder))
 	@{ \
 	set -e ;\
 	mkdir -p $(OTELCOL_BUILDER_DIR) ;\
-	curl -sLo $(OTELCOL_BUILDER) https://github.com/observatorium/opentelemetry-collector-builder/releases/download/v$(OTELCOL_BUILDER_VERSION)/opentelemetry-collector-builder_$(OTELCOL_BUILDER_VERSION)_linux_x86_64 ;\
+	curl -sLo $(OTELCOL_BUILDER) https://github.com/observatorium/opentelemetry-collector-builder/releases/download/v$(OTELCOL_BUILDER_VERSION)/opentelemetry-collector-builder_$(OTELCOL_BUILDER_VERSION)_linux_amd64 ;\
```
Comment (objectiser) [inline Makefile:13]: What is the reason for the change here?

Comment (jpkrohling) [reply Makefile:13]: This PR: https://github.com/observatorium/opentelemetry-collector-builder/pull/28

Comment (objectiser) [reply Makefile:13]: Ok thanks.

**Review by objectiser — APPROVED** [review]
